// nick_common_static/common/directives/schema_edit_directive.js
/**
 * Created by NICK on 15/12/3.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
define(["require", "exports"], function (require, exports) {
    //var json = require('../../node_modules/jsoneditor/dist/jsoneditor');
    var Directive = (function () {
        function Directive() {
        }
        Directive._name = "schemaEdit";
        Directive.directive = [function () {
                var dir = {
                    //compile?: IDirectiveCompileFn;
                    //controller?: any;
                    //controllerAs?: string;
                    //bindToController?: boolean|Object;
                    //link?: IDirectiveLinkFn | IDirectivePrePost;
                    //name?: string;
                    //priority?: number;
                    //replace?: boolean;
                    //require?: any;
                    //restrict?: string;
                    //scope?: any;
                    //template?: any;
                    //templateUrl?: any;
                    //terminal?: boolean;
                    //transclude?: any;
                    restrict: 'EA',
                    replace: false,
                    scope: {
                        json: '='
                    },
                    link: function ($scope, $element, $attrs) {
                        var _this = $scope;
                        _this['editor'] = new JSONEditor($element[0]);
                        $scope.$watch($attrs['json'], function () {
                            _this['editor'].set($scope['json']);
                        });
                    }
                };
                return dir;
            }];
        return Directive;
    })();
    exports.Directive = Directive;
});
