// nick_common_static/common/models/passport/bind_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
define(["require", "exports", "common/models/passport/user_model"], function (require, exports, um) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var Passport;
    (function (Passport) {
        /*
         * 用于绑定手机和邮箱
         * */
        var BindModel = (function (_super) {
            __extends(BindModel, _super);
            function BindModel(data) {
                return _super.call(this, data) || this;
            }
            return BindModel;
        }(um.Passport.UserModel));
        BindModel._schema = {
            type: 'object',
            description: '绑定邮箱和手机',
            required: ['email', 'mobile', 'smsToken'],
            properties: {
                email: {
                    type: 'string',
                    title: '邮箱地址',
                    format: 'email'
                },
                mobile: {
                    type: 'string',
                    title: '手机号码',
                    format: 'mobile'
                },
                smsToken: {
                    type: 'string',
                    title: '验证码'
                }
            }
        };
        Passport.BindModel = BindModel;
    })(Passport = exports.Passport || (exports.Passport = {}));
});
