// nick_common_static/common/models/passport/edit_password_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
define(["require", "exports", "common/models/common/base_data"], function (require, exports, bd) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var Passport;
    (function (Passport) {
        /*
         * 用于绑定手机和邮箱
         * */
        var EditPasswordModel = (function (_super) {
            __extends(EditPasswordModel, _super);
            function EditPasswordModel(data) {
                return _super.call(this, data) || this;
            }
            return EditPasswordModel;
        }(bd.Common.BaseData));
        EditPasswordModel._schema = {
            type: 'object',
            description: '修改密码',
            required: ['oldPassword', 'password', 'passwordAgain'],
            properties: {
                oldPassword: {
                    type: 'string',
                    title: '原始密码',
                    minLength: 6,
                    maxLength: 30
                },
                password: {
                    type: 'string',
                    title: '新密码',
                    minLength: 6,
                    maxLength: 30
                },
                passwordAgain: {
                    type: 'string',
                    title: '重复新密码',
                    minLength: 6,
                    maxLength: 30
                }
            }
        };
        Passport.EditPasswordModel = EditPasswordModel;
    })(Passport = exports.Passport || (exports.Passport = {}));
});
