// nick_common_static/common/models/passport/user_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
define(["require", "exports", "common/models/common/base_data"], function (require, exports, bd) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var Passport;
    (function (Passport) {
        /*
         * 用户模型
         * */
        var UserModel = (function (_super) {
            __extends(UserModel, _super);
            function UserModel(data) {
                return _super.call(this, data) || this;
            }
            return UserModel;
        }(bd.Common.BaseData));
        UserModel._schema = {
            type: 'object',
            description: '用户',
            required: ['nickname'],
            properties: {
                username: {
                    type: 'string',
                    title: '用户登录名'
                },
                nickname: {
                    type: 'string',
                    title: '昵称'
                },
                avatarUrl: {
                    type: 'string',
                    title: '头像'
                },
                mobile: {
                    type: 'string',
                    title: '手机'
                },
                email: {
                    type: 'string',
                    title: '邮箱'
                },
                userToken: {
                    type: 'string',
                    title: '用户标识'
                },
                password: {
                    type: 'string',
                    title: '密码'
                }
            }
        };
        Passport.UserModel = UserModel;
    })(Passport = exports.Passport || (exports.Passport = {}));
});
