// nick_common_static/common/models/brave/chapter_model.ts
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */

import cd = require('common/models/common/base_data');
import um = require('common/models/passport/user_model');

export module Brave {

    /*
     * 一日的勇者章节
     * */
    export class ChapterModel extends cd.Common.BaseData {
        public id:number;
        public tower:string;
        public num:number;
        public stageCount:number;
        public title:string;
        public description:string;
        public backgroundUrl:string;
        public selectBackgroundUrl:string;
        public aliveBackgroundUrl:string;
        public deadBackgroundUrl:string;
        public status:number;
        public aliveWords:string;
        public deadWords:string;
        public creator:um.Passport.UserModel;

        public static _convert(data:any) {
            var model = new Brave.ChapterModel({
                backgroundUrl: data.background_url,
                createAt: data.created_at,
                updateAt: data.updated_at,
                status: data.status,
                title: data.title,
                id: data.id,
                tower: data.tower,
                aliveWords: data.alive_words,
                deadWords: data.dead_words,
                num: data.num,
                stageCount: data.stages_count,
                description: data.description,
                selectBackgroundUrl: data.select_background_url,
                aliveBackgroundUrl: data.alive_background_url,
                deadBackgroundUrl: data.dead_background_url,
                creator: new um.Passport.UserModel({
                    nickname: data.nickname,
                    userToken: data.user_token,
                    avatarUrl: data.avatar_url
                })
            });

            return model;
        }

        public static _schema = {
            type: 'object',
            description: '章节列表',
            required: [
                'title',
                'tower',
                'num',
                'stageCount',
                'backgroundUrl'
            ],
            properties: {
                tower: {
                    type: 'string',
                    title: '塔'
                },
                num: {
                    type: 'number',
                    title: '第几章'
                },
                stageCount: {
                    type: 'number',
                    title: '总共多少关'
                },
                title: {
                    type: 'string',
                    title: '标题'
                },
                description: {
                    type: 'string',
                    title: '介绍'
                },
                backgroundUrl: {
                    type: 'string',
                    title: '背景地址'
                },
                aliveWords:{
                    type: 'string',
                    title: '存活文本'
                },
                deadWords:{
                    type: 'string',
                    title: '死亡文本'
                },
                selectBackgroundUrl: {
                    type: 'string',
                    title: '背景地址'
                },
                aliveBackgroundUrl: {
                    type: 'string',
                    title: '通关配图地址'
                },
                deadBackgroundUrl: {
                    type: 'string',
                    title: '死亡配图地址'
                }
            }
        };

        constructor(data?:any) {
            super(data);
        }
    }
}
