// nick_common_static/common/models/brave/chapter_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data', 'common/models/passport/user_model'], function (require, exports, cd, um) {
    var Brave;
    (function (Brave) {
        /*
         * 一日的勇者章节
         * */
        var ChapterModel = (function (_super) {
            __extends(ChapterModel, _super);
            function ChapterModel(data) {
                _super.call(this, data);
            }
            ChapterModel._convert = function (data) {
                var model = new Brave.ChapterModel({
                    backgroundUrl: data.background_url,
                    createAt: data.created_at,
                    updateAt: data.updated_at,
                    status: data.status,
                    title: data.title,
                    id: data.id,
                    tower: data.tower,
                    aliveWords: data.alive_words,
                    deadWords: data.dead_words,
                    num: data.num,
                    stageCount: data.stages_count,
                    description: data.description,
                    selectBackgroundUrl: data.select_background_url,
                    aliveBackgroundUrl: data.alive_background_url,
                    deadBackgroundUrl: data.dead_background_url,
                    creator: new um.Passport.UserModel({
                        nickname: data.nickname,
                        userToken: data.user_token,
                        avatarUrl: data.avatar_url
                    })
                });
                return model;
            };
            ChapterModel._schema = {
                type: 'object',
                description: '章节列表',
                required: [
                    'title',
                    'tower',
                    'num',
                    'stageCount',
                    'backgroundUrl'
                ],
                properties: {
                    tower: {
                        type: 'string',
                        title: '塔'
                    },
                    num: {
                        type: 'number',
                        title: '第几章'
                    },
                    stageCount: {
                        type: 'number',
                        title: '总共多少关'
                    },
                    title: {
                        type: 'string',
                        title: '标题'
                    },
                    description: {
                        type: 'string',
                        title: '介绍'
                    },
                    backgroundUrl: {
                        type: 'string',
                        title: '背景地址'
                    },
                    aliveWords: {
                        type: 'string',
                        title: '存活文本'
                    },
                    deadWords: {
                        type: 'string',
                        title: '死亡文本'
                    },
                    selectBackgroundUrl: {
                        type: 'string',
                        title: '背景地址'
                    },
                    aliveBackgroundUrl: {
                        type: 'string',
                        title: '通关配图地址'
                    },
                    deadBackgroundUrl: {
                        type: 'string',
                        title: '死亡配图地址'
                    }
                }
            };
            return ChapterModel;
        })(cd.Common.BaseData);
        Brave.ChapterModel = ChapterModel;
    })(Brave = exports.Brave || (exports.Brave = {}));
});
