// nick_common_static/common/models/brave/question_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data', 'common/models/passport/user_model'], function (require, exports, cd, um) {
    var Brave;
    (function (Brave) {
        var optionItem = (function (_super) {
            __extends(optionItem, _super);
            function optionItem(data) {
                _super.call(this, data);
            }
            return optionItem;
        })(cd.Common.BaseData);
        Brave.optionItem = optionItem;
        var QuestionOnModel = (function (_super) {
            __extends(QuestionOnModel, _super);
            function QuestionOnModel(data) {
                _super.call(this, data);
            }
            QuestionOnModel._schema = {
                type: 'object',
                description: '女神问题',
                required: [
                    'tower',
                    'chapter',
                    'stage'
                ],
                properties: {
                    tower: {
                        type: 'string',
                        title: '塔'
                    },
                    chapter: {
                        type: 'number',
                        title: '章节'
                    },
                    stage: {
                        type: 'number',
                        title: '关卡'
                    }
                }
            };
            return QuestionOnModel;
        })(cd.Common.BaseData);
        Brave.QuestionOnModel = QuestionOnModel;
        /*
         * 一日的勇者问题
         * */
        var QuestionModel = (function (_super) {
            __extends(QuestionModel, _super);
            function QuestionModel(data) {
                _super.call(this, data);
            }
            QuestionModel._convert = function (data) {
                var model = new Brave.QuestionModel({
                    backgroundUrl: data.background_url,
                    createAt: data.created_at,
                    updateAt: data.updated_at,
                    status: data.status,
                    title: data.title,
                    id: data.id,
                    tower: data.tower,
                    chapter: data.chapter_num,
                    stage: data.stage_num,
                    series: data.series,
                    question: data.question,
                    isRoot: data.is_root,
                    options: [],
                    creator: new um.Passport.UserModel({
                        nickname: data.nickname,
                        userToken: data.user_token,
                        avatarUrl: data.avatar_url
                    })
                });
                try {
                    data.options = JSON.parse(data.options);
                    data.options.forEach(function (d) {
                        model.options.push(new Brave.optionItem({
                            answer: d.answer,
                            is_ok: d.is_ok,
                            doa_words: d.doa_words,
                            words: d.words,
                            forward_to: d.forward_to,
                            piece: d.piece
                        }));
                    });
                }
                catch (e) {
                }
                return model;
            };
            QuestionModel._schema = {
                type: 'object',
                description: '女神问题',
                required: [
                    'title',
                    'content',
                    'location'
                ],
                properties: {
                    series: {
                        type: "string",
                        title: "系列"
                    },
                    question: {
                        type: 'string',
                        title: '问题'
                    },
                    backgroundUrl: {
                        type: 'string',
                        title: '配图地址'
                    },
                    isRoot: {
                        type: 'boolean',
                        title: '是否是根题目'
                    },
                    tower: {
                        type: 'string',
                        title: '塔'
                    },
                    chapter: {
                        type: 'number',
                        title: '章节'
                    },
                    stage: {
                        type: 'number',
                        title: '关卡'
                    },
                    title: {
                        type: 'string',
                        title: '标题'
                    },
                    options: {
                        type: 'array',
                        title: '选项',
                        items: {
                            type: 'object',
                            properties: {
                                answer: {
                                    type: 'string',
                                    title: '回答'
                                },
                                words: {
                                    type: 'string',
                                    title: '回答后的过渡文本'
                                },
                                doa_words: {
                                    type: 'string',
                                    title: '死亡或者存活文本'
                                },
                                forward_to: {
                                    type: 'number',
                                    title: '指向下一个问题id'
                                },
                                is_ok: {
                                    type: 'boolean',
                                    title: '是否通过',
                                    "default": true
                                },
                                piece: {
                                    type: 'string',
                                    title: '得到的称号片段'
                                }
                            }
                        }
                    }
                }
            };
            return QuestionModel;
        })(cd.Common.BaseData);
        Brave.QuestionModel = QuestionModel;
    })(Brave = exports.Brave || (exports.Brave = {}));
});
