// nick_common_static/common/models/brave/stage_model.js
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
define(["require", "exports", 'common/models/common/base_data'], function (require, exports, cd) {
    var Brave;
    (function (Brave) {
        var StageModel = (function (_super) {
            __extends(StageModel, _super);
            function StageModel(data) {
                _super.call(this, data);
            }
            StageModel._convert = function (data) {
                return new StageModel({
                    id: data.id,
                    tower: data.tower,
                    chapter: data.chapter_num,
                    stage: data.num,
                    questions: data.question_ids ? data.question_ids.split(',') : [],
                    crateAt: data.created_at,
                    updateAt: data.updated_at
                });
            };
            StageModel._schema = {
                type: 'object',
                description: '女神问题',
                required: [
                    'title',
                    'content',
                    'location'
                ],
                properties: {
                    tower: {
                        type: 'string',
                        title: '塔'
                    },
                    chapter: {
                        type: 'number',
                        title: '章节'
                    },
                    stage: {
                        type: 'number',
                        title: '关卡'
                    },
                    question_ids: {
                        type: 'string',
                        title: '问题IDS'
                    }
                }
            };
            return StageModel;
        })(cd.Common.BaseData);
        Brave.StageModel = StageModel;
    })(Brave = exports.Brave || (exports.Brave = {}));
});
