// nick_common_static/common/models/brave/stage_model.ts
/**
 * Created by NICK on 15/10/12.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */

import cd = require('common/models/common/base_data');
import um = require('common/models/passport/user_model');

export module Brave {

    export class StageModel extends cd.Common.BaseData {
        public questions:Array<number>;
        public id:number;
        public tower:string;
        public chapter:number;
        public stage:number;
        public question_ids:string;

        public static _convert(data:any):StageModel {
            return new StageModel({
                id: data.id,
                tower: data.tower,
                chapter: data.chapter_num,
                stage: data.num,
                questions: data.question_ids ? data.question_ids.split(',') : [],
                crateAt: data.created_at,
                updateAt: data.updated_at
            });
        }

        public static _schema = {
            type: 'object',
            description: '女神问题',
            required: [
                'title',
                'content',
                'location'
            ],
            properties: {
                tower: {
                    type: 'string',
                    title: '塔'
                },
                chapter: {
                    type: 'number',
                    title: '章节'
                },
                stage: {
                    type: 'number',
                    title: '关卡'
                },
                question_ids: {
                    type: 'string',
                    title: '问题IDS'
                }
            }
        };


        constructor(data ?:any) {
            super(data);
        }
    }
}
