// nick_common_static/common/config/config.js
/**
 * Created by NICK on 15/10/28.
 * email:nick121212@126.com
 * qq:289412378
 * copyright NICK
 */
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ProductConfig = (function () {
        function ProductConfig() {
            this.passport_api_url = 'http://106.15.72.133:6110/api';
            this.dashboard_api_url = 'http://106.15.72.133:6110/api';
            this.putong_api_url = 'http://106.15.72.133:6110/api';
            this.facemash_api = 'http://106.15.72.133:6110/api';
            this.shiji_api = 'http://106.15.72.133:6110/api';
            this.prefix = 'pro_';
            this.expire_in = 86400;
            this.app_token = 'iDnRcF7-';
        }
        return ProductConfig;
    }());
    var DevelopConfig = (function () {
        function DevelopConfig() {
            this.passport_api_url = 'http://192.168.1.244:8110/api';
            this.dashboard_api_url = 'http://192.168.1.244:8110/api';
            this.putong_api_url = 'http://192.168.1.244:8110/api';
            this.facemash_api = 'http://192.168.1.244:8110/api';
            this.shiji_api = 'http://192.168.1.244:8110/api';
            this.permiss_api = 'http://192.168.1.244:8110';
            this.prefix = 'dev_';
            this.app_token = 'iDnRcF7-';
            this.expire_in = 86400;
        }
        return DevelopConfig;
    }());
    exports.config = {
        pro: new ProductConfig(),
        dev: new DevelopConfig()
    };
});
